tests/python/test_matmul.py

```diff
@@ -201,3 +201,29 @@ def fusion_func(fd: FusionDefinition) -> None:
         ]
         outputs, _ = self.exec_nvfuser(fusion_func, inputs)
         assert outputs[0].ndim == 3
+
+    def test_matmul_stride(self):
+        n, h, l, s, e = 4, 8, 16, 16, 8
+        inputs = [
+            torch.randn(
+                n, h, l, e, device="cuda", dtype=torch.float16, requires_grad=True
+            ),
+            torch.randn(
+                n, h, s, e, device="cuda", dtype=torch.float16, requires_grad=True
+            ),
+        ]
+        for perm in itertools.permutations(range(4), 4):
+
+            def fusion_func(fd: FusionDefinition) -> None:
+                q = fd.from_pytorch(inputs[0])
+                k = fd.from_pytorch(inputs[1])
+                kt = fd.ops.permute(k, [0, 1, 3, 2])
+                out = fd.ops.matmul(q, kt)
```

Comment:
```suggestion
                k_t = fd.ops.permute(k, [0, 1, 3, 2])
                out = fd.ops.matmul(q, k_t)
```

Nit: `k_t` expresses better that it's a "transposition" of `k`.